docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+
+```shell
+dune --send-action hello addmsg '[ama, "good morning sunshine"]' hello@active
+dune --send-action hello addmsg '[hello, "good morning sunshine"]' hello@active
+```
+
+Find the `good morning sunshine` messages:
+
+```shell
+root@ubuntu-s-4vcpu-8gb-fra1-01:~# dune --send-action hello searchmsg '["good morning sunshine"]' hello@active
+```
+
+```txt
+executed transaction: dc8feaf0847b0c426f5e6c58bf532305e0625f617abf38555cc85d303bd82fd9  120 bytes  505 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::searchmsg             {"message":"good morning sunshine"}
+>> First message found. User: ama, Message: good morning sunshine
+>> Other message: User: hello, Message: good morning sunshine
+```
+
+## Authorization
+
+When a user or contract attempts to perform an action, the action can be validated by the EOS blockchain software. This validation process includes checking that the user or contract has the necessary permissions to perform the action. Permissions are granted through authorization, which involves associating a particular account or contract with a specific permission level.
+
+The `hello` contract does not perform any authorization checks. Any account could send any of its actions to the blockchain and they would be executed.
+
+Send `hi` action and sign it with the `hello@active` private key succeeds:
+
+```shell
+dune --send-action hello addmsg '[ama, "hi again"]' hello@active
+```
+
+```txt
+executed transaction: 4cee7cee2c76a764fde62c0f0719e138d9e1c3f69792f4a48b04beedacf8af64  112 bytes  453 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::addmsg                {"nm":"ama","message":"hi again"}
+```
+
+Send `hi` action and sign it with the `ama@active` private key succeeds:
+
+```shell
+dune --send-action hello addmsg '[ama, "hi again"]' ama@active
+```
+
+```txt
+executed transaction: a83c1623215c9d0be78a34e41a850716f176e5320fb4165879d7fab86599ff64  112 bytes  308 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::addmsg                {"nm":"ama","message":"hi again"}
+```
+
+You can implement an authorization check to permit only certain accounts or just on account to execute the `hello` contract's actions.
+
+### Use check() with has_auth()
+
+To perform an authorization check use the `check()` function in combination with the `has_auth` function. This combination enforces the action `addmsg` to be executed only by the account that is sent as first parameter, no matter what permission the account uses to sign the transaction (e.g. owner, active, code). If the check fails it raises an error with a custom message.
```

Comment:
Any reason this didn't use `require_auth`? It's always been more standard than a check with has_auth.